XenCert/StorageHandler.py

```diff
@@ -2981,6 +2981,327 @@ def MPConfigVerificationTests(self):
     def PoolTests(self):
         return (True, 1, 1)
 
+
+class StorageHandlerCIFS(StorageHandler):
+
+    def __init__(self, storage_conf):
+        XenCertPrint("Reached StorageHandlerCIFS constructor")
+        self.server = storage_conf['server']
+        self.username = storage_conf['username']
+        self.password = storage_conf['password']
+        StorageHandler.__init__(self, storage_conf)
+
+    def Create(self):
+        device_config = {}
+        device_config['server'] = self.server
+        device_config['username'] = self.username
+        device_config['password'] = self.password
+        retVal = True
+        try:
+            # Create an SR
+            Print("      Creating the SR.")
+            # try to create an SR with one of the LUNs mapped, if all fails throw an exception
+            XenCertPrint("The SR create parameters are %s, %s" % (util.get_localhost_uuid(self.session), device_config))
+            sr_ref = self.session.xenapi.SR.create(util.get_localhost_uuid(self.session), device_config, '0', 'XenCertTestSR', '', 'cifs', '',False, {})
+            XenCertPrint("Created the SR %s using device_config %s" % (sr_ref, device_config))
+            displayOperationStatus(True)
+
+        except Exception, e:
+            displayOperationStatus(False)
+            raise Exception(("   - Failed to create SR. Exception: %s" % str(e)))
+
+        if sr_ref == None:
+            displayOperationStatus(False)
+            retVal = False
+
+        return (retVal, sr_ref, device_config)
+
+    def __del__(self):
+        XenCertPrint("Reached StorageHandlerCIFS destructor")
+        StorageHandler.__del__(self)
+
+    def MetaDataTests(self):
+        Print("MetaDataTests not applicable to CIFS SR type.")
+        return False
+
+    def PoolTests(self):
+        try:
+            sr_ref = None
+            retVal = True
+            checkPoint = 0
+            totalCheckPoints = 0
+
+            totalCheckPoints += 4
+
```

Comment:
The tests below check for number of paths on each host in a pool based on multipathing. This doesn't make sense for CIFS SRs - they are not multipathed.